### PR TITLE
Yoks1729 image processing

### PR DESCRIPTION
**Pull request by Yoks1729**
This PR adds:

1. imgresize function has been created, which will take input of new height and width from the user,and will display the final edited image.

### CODE REVIEW

File: imgresize.cpp [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+{
+    // these variables aur used to store the dimensions of new image
+    double w,h;
+    // to store the dimensions of image
+    Size s = img.size();
+    cout<< "Resize Image." <<endl;
+    cout<< "Width: ";
+    cin>> w;
+    cout<< "Height: ";
+    cin>> h;
```
Comment (krshrimali) [inline imgresize.cpp]: Let's use `cout <<` and `cin >>` for consistency instead of `cout<<` and `cin>>`.

File: imgresize.cpp [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        h = h/s.height;
+    }
+    // this function resizes the image to our desire dimensions
+    resize (img,img,Size(),w,h);
+     // show our image inside it.
+    imshow (windowname,img);
+    // the int value inside wait key is time in milliseconds the program waits.
+    waitKey (500);
+}
+int main()
```
Comment (krshrimali) [inline imgresize.cpp]: Add a new line here, before `int main()`. Improves code readability.

File: imgresize.cpp [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+{
+    // variable s of string type is used to store the path of image which you want to display.
+    string s;
+    cout<< "Enter the path of Image which you want to display.";
+    cin>> s;
```
Comment (krshrimali) [inline imgresize.cpp]: Same here: `cout <<` and `cin >>` instead of `cout<<` and `cin>>`.

File: imgresize.cpp [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    img = imread(s);
+    // creating a window to display image.
+    namedWindow (windowname,WINDOW_AUTOSIZE);
+     // show our image inside it.
+    imshow (windowname,img);
+    // the int value inside wait key is time in milliseconds the program waits.
+    waitKey (300);
+    char key;
+    cout<< "Press any key amd enter to proceed " << endl;
+    cin>> key;
```
Comment (krshrimali) [inline imgresize.cpp]: Same space consistency here.

File: imgresize.cpp [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    while ( key != 'q')
+    {
+
+    cout<< "Enter 'r' to RESIZE and 'e' to EXIT" << endl ;
+    cin>> key;
```
Comment (krshrimali) [inline imgresize.cpp]: There is an extra space before semicolon here. Please remove that.

File: imgresize.cpp [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+        break;
+        case 'e':
+        // to exit the program. 
+         exit (0);
+         break;
+         default:
+         cout<< "Please enter a valid key" << endl;
+         break;
+        }
+    }
```
Comment (krshrimali) [inline imgresize.cpp]: This is better: (observe the indentation) :)

```
switch(key)
{
    case 'r':
        imgresize();
        break;
    case 'e':
        // to exit the program. 
        exit (0);
        break;
    default:
        cout<< "Please enter a valid key" << endl;
        break;
    }
}
```

File: imgresize.cpp [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    // to exit the program. 
+    exit (0);
```
Comment (krshrimali) [inline imgresize.cpp]: No need to add extra space here: `exit(0)` is fine. We don't add spaces for function calls.

File: imgshow.cpp [hunk 8]
```diff
@@ -0,0 +1,36 @@
+// cpp program to read and display the image using opencv. 
+#include <opencv2/opencv.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <iostream>
+
+using namespace std;
+using namespace cv;
+// declaraing global variables(so that can be used further in any of the function).
```
Comment (krshrimali) [inline imgshow.cpp]: An extra line after namespace will be good.

File: imgshow.cpp [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    string s;
+    cout<< "Enter the path of Image which you want to display.";
+    cin>> s;
+ // imread returns the image data in the array.
+    img = imread(s);
+ // creating a window to display image.
+    namedWindow (windowname,WINDOW_AUTOSIZE);
+ // show our image inside it
+    imshow (windowname,img);
+ // the int value inside wait key is time in milliseconds the program waits.
```
Comment (krshrimali) [inline imgshow.cpp]: The comments need to be indented, to make code look beautiful ;) Also, remove spaces for function calls:

```
namedWindow(windowname, WINDOW_AUTOSIZE);
// show our image inside it
imshow(windowname, img);
```
is better!

File: imgshow.cpp [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+    waitKey (300);
+    char k;
+    cout<< "Press 'q' to EXIT" << endl;
+    cin>> k;
+    if (k=='q')
+    {
+    // to exit the program.
```
Comment (krshrimali) [inline imgshow.cpp]: Indent the comment.

**Review by krshrimali — CHANGES_REQUESTED** [review]
Left some comments for code readability. Good work so far.

**Comment by Yoks1729** [comment]
> Left some comments for code readability. Good work so far.

Thank you very much sir , u really helped me a lot.

File: imgresize.cpp [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+    // creating a window to display image.
+    namedWindow(windowname, WINDOW_AUTOSIZE);
+     // show our image inside it.
+    imshow(windowname, img);
+    // the int value inside wait key is time in milliseconds the program waits.
+    waitKey(300);
+    char key;
+    cout << "Press any key amd enter to proceed " << endl;
+    cin >> key;
+    while( key != 'q')
```
Comment (krshrimali) [inline imgresize.cpp]: `while(key != 'q')` -- This!

**Review by krshrimali — CHANGES_REQUESTED** [review]

File: imgresize.cpp (hunk 11 above)
Comment (Yoks1729) [reply imgresize.cpp]: done

**Pull request merged**